### PR TITLE
generate host_config_key using random UUIDs, not a time-based md5 hash

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

**Comment by ryanpetrello** [comment]
see: https://github.com/ansible/tower/issues/814

File: awx/ui/client/src/templates/job_templates/factories/md-5-setup.factory.js [hunk 1]
```diff
@@ -11,7 +11,9 @@ export default
 
             scope.genMD5 = function (fld) {
                 var now = new Date();
-                scope[fld] = md5.createHash('AnsibleWorks' + now.getTime());
+                scope[fld] = ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+                    (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
```
Comment (ryanpetrello) [inline awx/ui/client/src/templates/job_templates/factories/md-5-setup.factory.js:14]: https://caniuse.com/#feat=getrandomvalues

**Comment by ryanpetrello** [comment]
kapture 2018-11-16 at 16 49 19

File: awx/ui/client/src/templates/job_templates/factories/md-5-setup.factory.js [hunk 2]
```diff
@@ -26,5 +27,3 @@ export default
             };
         };
     }
-
-md5Setup.$inject = [   'md5'   ];
```
Comment (wenottingham) [inline awx/ui/client/src/templates/job_templates/factories/md-5-setup.factory.js:30]: can probably remove some md5 libs now

Comment (ryanpetrello) [reply awx/ui/client/src/templates/job_templates/factories/md-5-setup.factory.js:30]: That'd be nice.  I dunno how to do that 🤷‍♂️

File: awx/ui/client/src/templates/job_templates/factories/md-5-setup.factory.js [hunk 3]
```diff
@@ -10,8 +10,9 @@ export default
             master[check_field] = default_val;
 
             scope.genMD5 = function (fld) {
-                var now = new Date();
-                scope[fld] = md5.createHash('AnsibleWorks' + now.getTime());
+                scope[fld] = ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
```
Comment (kialam) [inline awx/ui/client/src/templates/job_templates/factories/md-5-setup.factory.js:13]: This line of code is truly outstanding. A comment linking to the [original gist](https://gist.github.com/jed/982883) might be helpful for those curious about how this part works.

**Comment by ryanpetrello** [comment]
Replaced by 
https://github.com/ansible/awx/pull/2747

**Pull request closed without merging**